ddtrace/tracer/spancontext.go

```diff
@@ -187,6 +227,9 @@ func (t *trace) ackFinish() {
 	}
 	if tr, ok := internal.GetGlobalTracer().(*tracer); ok {
 		// we have a tracer that can receive completed traces.
+		if t.priority != nil {
+			t.root.Metrics[keySamplingPriority] = *t.priority
```

Comment:
As long as there is a span in the trace, there is a root, and since this method is only called from a span that is part of the trace, root can not be nil at this stage.